code/game/mecha/mecha_defense.dm

```diff
@@ -113,12 +113,29 @@
 	. = ..()
 
 /obj/mecha/ex_act(severity, target)
+	contents_explosion(severity++, target)
 	log_message("Affected by explosion of severity: [severity].",1)
 	if(prob(deflect_chance))
 		severity++
 		log_append_to_last("Armor saved, changing severity to [severity].")
 	. = ..(severity, target)
 
+/obj/mecha/contents_explosion(severity, target)
+	for(var/X in equipment)
+		var/obj/item/mecha_parts/mecha_equipment/ME = X
+		ME.ex_act(severity,target)
+	for(var/Y in trackers)
+		var/obj/item/mecha_parts/mecha_tracking/MT = Y
+		MT.ex_act(severity, target)
+	if(occupant)
+		occupant.ex_act(severity,target)
+
+/obj/mecha/handle_atom_del(atom/A)
+	if(A == occupant)
+		occupant = null
+		icon_state = initial(icon_state)+"-open"
+		setDir(dir_in)
```

Comment:
if the occupant is gibbed while inside, the mech updates its appearance to look open.